src/clp_logging/readers.py

```diff
@@ -186,7 +192,7 @@ def read_preamble(self) -> int:
             # We do not use the timestamp pattern from the preamble as it may
             # be from other languages and therefore incompatible.
             # self.timestamp_format = self.metadata[METADATA_TIMESTAMP_PATTERN_KEY]
-            self.timezone = dateutil.tz.gettz(self.metadata[METADATA_TZ_ID_KEY])
+            self.timezone = ZoneInfo(self.metadata[METADATA_TZ_ID_KEY])
```

Comment:
The class `ZoneInfo` is an implementation of `tzinfo` abstracted class, shall we narrow down the type of `self.timezone` to `ZoneInfo` instead?
This means we should also update the type annotation in `Log._decode`